core/services/gateway/handlers/functions/orm.go

```diff
@@ -0,0 +1,135 @@
+package functions
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/functions/generated/functions_router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+//go:generate mockery --quiet --name ORM --output ./mocks/ --case=underscore
+
+type ORM interface {
+	GetSubscriptions(offset, limit uint, qopts ...pg.QOpt) ([]CachedSubscription, error)
+	UpsertSubscription(subscription CachedSubscription, qopts ...pg.QOpt) error
+}
+
+type orm struct {
+	q pg.Q
+}
+
+var _ ORM = (*orm)(nil)
+var (
+	ErrDuplicateSubscriptionID = errors.New("Functions ORM: duplicate subscription ID")
+	ErrInvalidParameters       = errors.New("invalid parameters provided to create a subscription cache ORM")
+)
+
+const (
+	tableName = "functions_subscriptions"
+)
+
+type cachedSubscriptionRow struct {
+	SubscriptionID        uint64
+	Balance               string
+	Owner                 common.Address
+	BlockedBalance        string
+	ProposedOwner         common.Address
+	Consumers             pq.ByteaArray
+	Flags                 []uint8
+	RouterContractAddress common.Address
+}
+
+func NewORM(db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig) (ORM, error) {
```

Comment:
RouterContractAddress is constant for every ORM instance so another option is to pass it in the constructor and store for every use. That way you could avoid including it inside CachedSubscription. Up to you.

Reply:
good call, its cleaner